### ISSUE TITLE
Netty accepts invalid requests with NUL between the CR and LF following the request line

### ISSUE THREAD

**Issue by kenballus** [netty/netty#14166]
### Steps to reproduce
1. Start any Netty HTTP/1.1 server on `127.0.0.1:8000`.
2. Send it a request with a NUL between the request-line's CR and LF. For example,
```sh
printf 'GET / HTTP/1.1\r\x00\nHost: whatever\r\n\r\n' | nc localhost 8000
```
3. Observe that Netty responds to the invalid request.

### Expected behavior
Netty should close the connection because the request is invalid.

### Actual behavior
Netty responds to the request as though the NUL weren't there.

### Netty version
4.1 @ 967be496c6

### JVM version (e.g. `java -version`)
```
openjdk version "23-ea" 2024-09-17
OpenJDK Runtime Environment (build 23-ea+20-Debian-1)
OpenJDK 64-Bit Server VM (build 23-ea+20-Debian-1, mixed mode, sharing)
```

### OS version (e.g. `uname -a`)
```
Linux 416cc96d1b8b 6.9.7-arch1-1 #1 SMP PREEMPT_DYNAMIC Fri, 28 Jun 2024 04:32:50 +0000 x86_64 GNU/Linux
```

**Comment by normanmaurer** [comment]
Should be fixed by https://github.com/netty/netty/pull/14180

**Comment by kenballus** [comment]
Confirmed that this is fixed by #14180. Thanks!
